### PR TITLE
Modify the GitHub Action of setting up the Minikube cluster

### PR DESCRIPTION
**Pull request by runzhliu**
### _Why are the changes needed?_

bump the minikube version to 1.29.0 and the kubernetes version to 1.26.1.

After ran some tests for the used action manusa/actions-setup-minikube@v2.7.2, it turns out that it would have some problems when I bump the minikube version to 1.29.0, please refer to [this link](https://github.com/runzhliu/yum-with-browser/actions/runs/4113082204/jobs/7098775474) for details.

In addition, I have tried another minikube action medyagh/setup-minikube@master, it's good to bump the version for both minikube  and Kubernetes, please refer to [this link](https://github.com/runzhliu/yum-with-browser/actions/runs/4112397435/jobs/7097254604#step:5:8).

However, I would prefer to [this way](https://github.com/apache/spark/blob/master/.github/workflows/build_and_test.yml#L945-L966) to setup a minikube cluster for the rest Kubernetes related tests, it's also good to set up the cluster as well, please refer to [this link](https://github.com/runzhliu/yum-with-browser/actions/runs/4113018527/jobs/7098637193#step:4:2).

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Review by bowenliang123 — APPROVED** [review]

**Comment by bowenliang123** [comment]
Seems some error encountered in CI. 
https://github.com/apache/kyuubi/actions/runs/4099996056/jobs/7072995365#step:5:110

Shall we bump minikube first and the a followup for bumping kubernetes?

**Comment by runzhliu** [comment]
> Seems some error encountered in CI. https://github.com/apache/kyuubi/actions/runs/4099996056/jobs/7072995365#step:5:110
> 
> Shall we separate them as bumping minikube first and then the kubernetes? @runzhliu


Sure, so I would first bump the version of Minukube in this PR, I'll also take some time to dig into this problem.

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by runzhliu** [comment]
image
Minikube's Node seems to be not ready, I would run some tests for this version.

**Review by bowenliang123 — CHANGES_REQUESTED** [review]
What's the reason for changeing the used action for minikube ?

**Comment by runzhliu** [comment]
> What's the reason for changeing the used action for minikube ?

Run some tests for the used action and another minikube action, assuming that the used one is not run the full test for the higher version of Kubernetes, it might be the reason for CI failed.
In this case, I prefer to [this way](https://github.com/apache/spark/blob/master/.github/workflows/build_and_test.yml#L945-L966) to setup a minikube cluster for the rest Kubernetes related tests. @bowenliang123

**Comment by bowenliang123** [comment]
Try to update the PR title and description with the latest progress and related purposed plan or details. You have tried two different ways for setting up the minikube , the first is `medyagh/setup-minikube` action in the latest commit, the second is the link to Spark with manual installation of minikube.

**Comment by runzhliu** [comment]
> Try to update the PR title and description with the latest progress and related purposed plan or details. You have tried two different ways for setting up the minikube , the first is `medyagh/setup-minikube` action in the latest commit, the second is the link to Spark with manual installation of minikube.

OK.

**Comment by bowenliang123** [comment]
Are you using the preferred way as you mentioned in description? I didn't get it.

**Comment by pan3793** [comment]
SGTM to install minikube manually like spark does, but please use the specific version instead of latest

**Comment by runzhliu** [comment]
> 

Sorry to make a mistake, Not at the time when you commented.

Now I just made another commit and trigger the CI process to see if would work.

**Comment by bowenliang123** [comment]
No problem. I agree with @pan3793 , it's alright to install and use minikube manually without actions as long as in a proper way like Spark does.

**Review by runzhliu — COMMENTED** [review]
bump the minikube version to 1.29.0 and the kubernetes version to 1.26.1

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -394,15 +394,12 @@ jobs:
...
-        uses: manusa/actions-setup-minikube@v2.7.2
-        with:
-          minikube version: 'v1.28.0'
-          kubernetes version: 'v1.25.4'
-          github token: ${{ secrets.GITHUB_TOKEN }}
-          driver: docker
-          start args: '--extra-config=kubeadm.ignore-preflight-errors=NumCPU --force --cpus 2 --memory 4096'
+        run: |
+          # https://minikube.sigs.k8s.io/docs/start/
+          curl -LO https://github.com/kubernetes/minikube/releases/download/v1.29.0/minikube-linux-amd64
```
Comment (pan3793) [inline .github/workflows/master.yml:400]: let's pullout `KUBERNETES_VERSION` and `MINIKUBE_VERSION` to L36

Comment (runzhliu) [reply .github/workflows/master.yml:400]: Good point!

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -347,15 +349,16 @@ jobs:
...
-        uses: manusa/actions-setup-minikube@v2.7.2
-        with:
-          minikube version: 'v1.28.0'
-          kubernetes version: 'v1.25.4'
-          github token: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # https://minikube.sigs.k8s.io/docs/start/
+          curl -LO https://github.com/kubernetes/minikube/releases/download/${MINIKUBE_VERSION}/minikube-linux-amd64
+          sudo install minikube-linux-amd64 /usr/local/bin/minikube
+          minikube start --cpus 2 --memory 4096 --kubernetes-version=${KUBERNETES_VERSION} --force
```
Comment (cfmcgrady) [inline .github/workflows/master.yml:357]: add a comment to explain why we are doing this resource limit.

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -347,15 +349,16 @@ jobs:
...
+          minikube image load apache/kyuubi:latest
```
Comment (cfmcgrady) [inline .github/workflows/master.yml:358]: add a comment to explain why we need to load the local docker image.

**Review by bowenliang123 — APPROVED** [review]
LGTM

File: .github/workflows/master.yml [hunk 4]
```diff
@@ -394,15 +398,12 @@ jobs:
...
-        uses: manusa/actions-setup-minikube@v2.7.2
-        with:
-          minikube version: 'v1.28.0'
-          kubernetes version: 'v1.25.4'
-          github token: ${{ secrets.GITHUB_TOKEN }}
-          driver: docker
-          start args: '--extra-config=kubeadm.ignore-preflight-errors=NumCPU --force --cpus 2 --memory 4096'
+        run: |
+          # https://minikube.sigs.k8s.io/docs/start/
+          curl -LO https://github.com/kubernetes/minikube/releases/download/${MINIKUBE_VERSION}/minikube-linux-amd64
```
Comment (bowenliang123) [inline .github/workflows/master.yml:404]: Cache the minikube in a followup PR to avoid file downloading in every build.

**Review by pan3793 — APPROVED** [review]

**Review by cfmcgrady — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master/1.7 .

**Pull request closed without merging**